source/create.ts

```diff
@@ -16,52 +51,55 @@ const aliases = [
 	'delete'
 ];
 
-const create = defaults => {
-	defaults = merge({}, defaults);
-	preNormalizeArguments(defaults.options);
+const create = (defaults: Partial<Defaults>): Got => {
+	defaults = merge<Defaults, Partial<Defaults>>({}, defaults);
+	preNormalizeArguments(defaults.options!);
 
 	if (!defaults.handler) {
 		// This can't be getPromiseOrStream, because when merging
 		// the chain would stop at this point and no further handlers would be called.
 		defaults.handler = (options, next) => next(options);
 	}
 
-	function got(url, options?: any) {
+	// @ts-ignore Because the for loop handles it for us, as well as the other Object.defines
+	const got: Got = (url: URL | string | Partial<Options>, options?: Partial<Options>): ProxyStream | CancelableRequest<Response> => {
 		try {
-			return defaults.handler(normalizeArguments(url, options, defaults), getPromiseOrStream);
+			return defaults.handler!(normalizeArguments(url, options as NormalizedOptions, defaults), getPromiseOrStream);
```

Comment:
This doesn't seem right. `options` aren't normalized yet so I have no idea why you're calling `options as NormalizedOptions` here. `defaults` are. Any idea?

Reply:
That relates to https://github.com/sindresorhus/got/pull/797#issuecomment-491620160, and depending on that outcome, will change accordingly